src/navigation/courses/components/course-breadcrumbs/index.js

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import { Breadcrumb } from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+import { domain_api, language_api, soft_skill_api } from '../../../../common-services/api-endpoints';
+import { DOMAINS, LANGUAGES, SOFT_SKILLS } from '../../../../common-services/course_types';
+
+export default class CourseBreadcrumbs extends Component {
+  state = { categoryName: '' };
+
+  technical_breadcrumb = [
+    {
+      name: 'Technial Skills',
+      path: '/technical',
+      active: false
+    }
+  ];
+
+  random_data_breadcrumbs = [
+    ...this.technical_breadcrumb,
+    {
+      name: 'Miscellaneous',
+      path: '/technical/misc',
+      active: true
+    }
+  ]
+
+  knowledege_base_breadcrumbs = [
+    ...this.technical_breadcrumb,
+    {
+      name: 'Knowledge Base',
+      path: '/technical/knowledge-base',
+      active: false
+    }
+  ];
+
+  domain_breadcrumbs = [
+    ...this.knowledege_base_breadcrumbs,
+    {
+      name: 'Domains',
+      path: '/technical/knowledge-base/domains',
+      active: false
+    }
+  ]
+  language_breadcrumbs = [
+    ...this.knowledege_base_breadcrumbs,
+    {
+      name: 'Languages',
+      path: '/technical/knowledge-base/languages',
+      active: false
+    }
+  ]
+  soft_skills_breadcrumbs = [
+    {
+      name: 'Soft Skills',
+      path: '/soft-skills'
+    }
+  ]
+
+  componentDidMount() {
+    let { courseType, categoryId } = this.props;
+    this.fetchCourseType(courseType, categoryId);
+  }
+  componentWillReceiveProps(nextProps) {
+    let { courseType, categoryId } = nextProps;
+    if(courseType !== this.props.courseType || categoryId !== this.props.categoryId) {
+      this.fetchCourseType(courseType, categoryId);
+    }
+  }
+
+  fetchCourseType = (courseType, categoryId) => {
+    this.setState({ categoryName: ''});
+    // courseType & courseId passed down as props;
+    let url = '';
+    let name_attribute = '';
+    let categoryName = '';
+
+    if(courseType === DOMAINS) {
+      url = domain_api;
+      categoryName = 'domain_name';
+    }
+    if(courseType === LANGUAGES) {
+      url = language_api;
+      categoryName = 'language_name';
+    }
+    if(courseType === SOFT_SKILLS) {
+      url = soft_skill_api;
+      categoryName = 'soft_skill_category';
+    }
+    fetch(`${url}${categoryId}`)
+    .then(response => {
+      response.json()
+      .then(data => {
+        if(data[categoryName]) {
+          this.setState({ categoryName: data[categoryName] });
+        }
+      })
```

Comment:
Instead of randomly calling fetch here and there and missing out a few catches, create a service method (like apiCall or something) and call fetch from inside it. So in your components you'll just have to do (`apiCall('get', 'url_slug').then(results...)`). It is also more maintainable.